lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
-
   const minimalProps = {
-        label: 'dropdown',
-        isOpen: false,
-        onToggleCollapse: () => {}
-      },
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxDropdown, minimalProps),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxDropdown, minimalProps);
-
-  beforeEach(function() {
-    // Avoid rendering directly on the body.
-    container = document.createElement('div');
-    document.body.appendChild(container);
-  });
+    label: 'dropdown-toggle',
+    isOpen: false,
+    onToggleCollapse: () => {}
+  };
 
-  afterEach(function() {
-    if (container) {
-      document.body.removeChild(container);
-      container = null;
-    }
-  });
-
-  it('renders a button with the appropriate classes and type=button', function() {
-    const component = getMountedComponent();
-    const dropdown = component.find(AbstractDropdown);
-    const button = dropdown.find(NxButton);
-    const icon = button.find(NxFontAwesomeIcon);
-
-    expect(dropdown).toHaveClassName('.nx-dropdown');
-    expect(button).toHaveClassName('.nx-dropdown__toggle');
-    expect(button).toHaveProp('variant', 'tertiary');
-    expect(button).toHaveProp('type', 'button');
-    expect(button.childAt(0)).toContainReact(<span className="nx-dropdown__toggle-label">dropdown</span>);
-    expect(icon).toHaveProp('icon', faCaretDown);
-  });
+  const quickRender = rtlRender(NxDropdown, minimalProps);
+  const renderEl = rtlRenderElement(NxDropdown, minimalProps);
 
   it('renders the button according to the supplied variant', function() {
-    let component = getMountedComponent();
-    expect(component.find(NxButton)).toHaveProp('variant', 'tertiary');
+    const { rerender } = quickRender();
+
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--tertiary');
 
-    component = getMountedComponent({ variant: 'primary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'primary');
+    rerender(<NxDropdown {...minimalProps} variant="primary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--primary');
 
-    component = getMountedComponent({ variant: 'secondary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'secondary');
+    rerender(<NxDropdown {...minimalProps} variant="secondary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--secondary');
 
-    component = getMountedComponent({ variant: 'error' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'error');
+    rerender(<NxDropdown {...minimalProps} variant="error" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--error');
   });
 
   it('correctly renders the menu based on isOpen prop', function() {
-    let component = getMountedComponent({ isOpen: true });
-    expect(component.find(NxDropdownMenu)).toExist();
+    const { container, rerender } = quickRender({ ...minimalProps, isOpen: true });
+    const menu = container.querySelector('.nx-dropdown-menu');
```

Comment:
If `nx-dropdown-menu` had proper aria roles, I'd say you should use those.  I'm assuming the reason you're using the classname is because it doesn't have any roles.  A comment should be added pointing that out and referring to the JIRA ticket where that should be fixed (create the ticket if we don't already have one).

Reply:
> If `nx-dropdown-menu` had proper aria roles, I'd say you should use those. I'm assuming the reason you're using the classname is because it doesn't have any roles. A comment should be added pointing that out and referring to the JIRA ticket where that should be fixed (create the ticket if we don't already have one).

Added comment here: 9389efa4ca4eb23eabe01dab16a11cf70bc0f225